hasktorch/src/Torch/Typed/Functional.hs

```diff
@@ -1852,11 +1852,12 @@ constantPadNd1d value input = unsafePerformIO $ ATen.cast3
 type ConvSideCheck (inputSize :: Nat) (kernelSize :: Nat) (stride :: Nat) (padding :: Nat) (outputSize :: Nat) =
   (
     -- kernel size and stride must be > 0
-    kernelSize >= 1, stride >= 1
+    1 <= kernelSize, 1 <= stride
     -- kernel size can't be greater than actual input size
-  , ((inputSize + (2 * padding)) + 1) >= kernelSize
+    -- ToDo: Do not use '>=' on constraint to avoid reduction-stack-overflow.
+  , (kernelSize - 1) <= (inputSize + (2 * padding))
```

Comment:
This `>=` seems like an obstacle.
When we use `>=` , deduction-error occurs.

Reply:
In addition to this `>=`, `  , `kernelSize <= (inputSize + (2 * padding) + 1)` becomes deduction-error too, so I use `(kernelSize -1) <= (inputSize + (2 * padding))` .

Reply:
Not ghc-typelits-natnormalise-0.7.2 but 0.7 allows the deduction-error.(I do not know why.)